### ISSUE TITLE
[TASK] Implement data-driven optimal block length selection for MCS Block-Bootstrap

### ISSUE THREAD

**Issue by Torim98** [Torim98/regime-switching-daa#22]
## Objective

The Monte Carlo Simulation (MCS) in `05_evaluation.ipynb` currently uses a hardcoded block length of `block_length: 20` (configured in `config.yaml` under `evaluation.mcs`). This value was chosen heuristically without formal justification.

The block length is a critical parameter in the Block-Bootstrap: short blocks destroy long-range dependencies in the return series, while overly long blocks reduce path variability and the number of independent blocks in the sample. Politis & White (2004) propose a data-driven procedure for optimal block length selection that adapts to the autocorrelation structure of the time series, removing the need for manual tuning.

Replacing the static value with an automatically determined block length strengthens the statistical validity of the MCS results and provides a formally justified parameter choice for the thesis (Chapter 3.5.6).

## Scope

**In scope:**
- Implement optimal block length estimation using the Politis & White (2004) method (with the Patton, Politis & White (2009) correction)
- Replace the static `block_length: 20` with the dynamically computed value at runtime
- Log / print the computed block length for reproducibility
- Document the method and its output in the thesis context

**Out of scope:**
- Changing the bootstrap method itself (Circular vs. Stationary; keep current approach)
- Comparing multiple block length estimators

## Plan

1. Add `arch` package (by Kevin Sheppard) as a dependency in `00_dependencies.ipynb`; it provides `arch.bootstrap.optimal_block_length`, a mature implementation of the Politis-White algorithm including the 2009 correction
2. In `05_evaluation.ipynb`, before the MCS loop, compute the optimal block length:
   ```python
   from arch.bootstrap import optimal_block_length
   opt_bl = optimal_block_length(returns_series)
   block_length = int(round(opt_bl.loc[returns_series.name, "circular"]))
   # or "stationary" depending on the bootstrap variant used

**Dependencies:**
- Potential interaction with #2 (Optuna hyperparameter optimization): If Optuna optimizes parameters that affect the return series used for MCS (e.g., `window_size`, signal threshold), the optimal block length may change between optimization runs. The block length computation should therefore run **after** hyperparameters are finalized; i.e., it must be part of the evaluation step (`05_evaluation`), not the training step (`03_regime_switching_models`). If #2 introduces walk-forward validation with multiple folds, consider whether the block length should be recomputed per fold or fixed once on the full test set.

**Comment by Torim98** [comment]
Closing as out of scope.

The proposed changes exceed the scope of the current project. Therefore, this feature will not be pursued further, and the issue is closed for documentation purposes.
